src/workerd/api/trace.c++

```diff
@@ -142,141 +277,112 @@ jsg::Dict<jsg::ByteString, jsg::ByteString> TraceItem::FetchEventInfo::Request::
 }
 
 kj::StringPtr TraceItem::FetchEventInfo::Request::getMethod() {
-  return kj::toCharSequence(eventInfo.method);
+  return detail->method;
 }
 
 kj::String TraceItem::FetchEventInfo::Request::getUrl() {
-  kj::StringPtr url = eventInfo.url;
-  return (redacted ? redactUrl(url) : kj::str(url));
+  return (redacted ? redactUrl(detail->url) : kj::str(detail->url));
 }
 
 jsg::Ref<TraceItem::FetchEventInfo::Request> TraceItem::FetchEventInfo::Request::getUnredacted() {
-  auto request = jsg::alloc<Request>(kj::addRef(*trace), eventInfo);
-  request->redacted = false;
-  return kj::mv(request);
+  return jsg::alloc<Request>(*detail);
```

Comment:
I guess it's probably less code and maybe slightly more efficient to do it this way, and probably a moot point with sufficient test coverage...  But it does feel a little error-prone to have a constructor overload with an insecure default.

I assume it's not possible to mark the unredacted constructor as private, due to compatibility with `jsg::alloc()` internals?  If not, I'd almost rather have the explicit parameter, or the explicit field setting, just to make it clear from the call site that this is the only place redaction gets set.

Hrm...  I'm not totally familiar with `JSG_LAZY_READONLY_INSTANCE_PROPERTY()`, but I suppose it does caching, implying that the fields it is based on should be constant.  But I suppose actually marking them const wouldn't provide much benefit.

Reply:
> I'd almost rather have the explicit parameter, or the explicit field setting...

That's fair.

> I'm not totally familiar with JSG_LAZY_READONLY_INSTANCE_PROPERTY, but I suppose it does caching...

Yeah, when the property is first accessed v8 will call the backing c++ implementation. Once a value is provided, it will replace the defined property config with the value such that subsequent accesses will use the cached value rather than invoking the backing c++ function again. There's really no need to mark the properties const at the c++ level but it also doesn't harm anything to do so.

Reply:
Ok, updated to pass in the `redacted` argument explicitly.